http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/utils/NettyUtils.java

```diff
@@ -403,6 +406,20 @@ public static boolean isAlpnSupported(SslProvider sslProvider) {
             return true;
         }
 
+        Boolean supported = alpnSupported;
+        if (supported != null) {
+            return supported;
+        }
+
+        synchronized (NettyUtils.class) {
+            if (alpnSupported == null) {
+                alpnSupported = checkAlpnSupport();
+            }
```

Comment:
We need to double check the value here. Can we use Lazy directly, which has built-in lazy initialization support?

https://github.com/aws/aws-sdk-java-v2/blob/master/utils/src/main/java/software/amazon/awssdk/utils/Lazy.java